### PR TITLE
[Spec] Completed the spec of `from_bcs`, `any`, `copyable_any` and `type_info`

### PR DESCRIPTION
**Pull request by junkil-park**
### Description

Added the spec of the following modules:
- `from_bcs`
- `any`
- `copyable_any`
- `type_info`

There are TODOs which are filed as separate Github issues:
- https://github.com/move-language/move/issues/699
- https://github.com/move-language/move/issues/700
- https://github.com/move-language/move/issues/701
- https://github.com/move-language/move/issues/702

### Test Plan
aptos move prove

### CODE REVIEW

File: aptos-move/framework/aptos-stdlib/sources/type_info.spec.move [hunk 1]
```diff
@@ -1,8 +1,28 @@
 spec aptos_std::type_info {
     // Move Prover natively supports `type_of` and `type_name`.
 
-    spec chain_id_internal {
-        // TODO: temporary mockup.
+    spec fun spec_is_primitive_type<T>(): bool {
```
Comment (meng-xu-cs) [inline aptos-move/framework/aptos-stdlib/sources/type_info.spec.move:4]: Do not use this approach to test whether a `T` is not a struct, instead, defined a spec function
```
spec native fun spec_is_struct<T>(): bool;
```

See the test case for an example:
https://github.com/move-language/move/blob/main/language/move-prover/tests/sources/functional/type_reflection.move#L13

**Review by meng-xu-cs — CHANGES_REQUESTED** [review]

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-stdlib/sources/type_info.spec.move (hunk 1 above)
Comment (junkil-park) [reply aptos-move/framework/aptos-stdlib/sources/type_info.spec.move]: Thanks! Updated using `spec_is_struct`.

File: aptos-move/framework/aptos-token/doc/token.md [hunk 2]
```diff
@@ -138,13 +138,15 @@ Checkout our developer doc on our token standard https://aptos.dev/concepts/coin
 <code>amount: u64</code>
 </dt>
 <dd>
-
+ the amount of tokens. Only property_version = 0 can have a value bigger than 1.
```
Comment (meng-xu-cs) [inline aptos-move/framework/aptos-token/doc/token.md:141]: Why we can an update on this auto-generated file?

Comment (junkil-park) [reply aptos-move/framework/aptos-token/doc/token.md]: Good call! This change has been removed from this commit when I rebased it onto the head.

**Review by meng-xu-cs — APPROVED** [review]

**Pull request merged**